src/overlays/actors/ovl_En_Encount2/z_en_encount2.c

```diff
@@ -22,19 +36,311 @@ const ActorInit En_Encount2_InitVars = {
+                if ((Rand_ZeroFloat(1.99f) < 1.0f) && (LINK_IS_CHILD)) {
+                    // rock spawn pos X, Z near player
+                    tempVec2X = Rand_CenteredFloat(10.0f) + player->actor.world.pos.x;
+                    tempVec2Z = Rand_CenteredFloat(10.0f) + player->actor.world.pos.z;
+                } else {
+                    if (player->linearVelocity != 0.0f) {
+                        // rock spawn pos is between 300 and 600 units from the camera depending on the camera yaw.
+                        // Rocks will generally spawn closer to the camera in the X axis than in the Z axis.
+                        tempVec2X = Rand_CenteredFloat(200.0f) + (globalCtx->view.eye.x + (tempVec2X * 300.0f));
+                        tempVec2Z = Rand_CenteredFloat(50.0f) + (globalCtx->view.eye.z + (tempVec2Z * 600.0f));
+                    } else {
+                        // rock spawn pos X, Z near player
+                        tempVec2X = Rand_CenteredFloat(10.0f) + player->actor.world.pos.x;
+                        tempVec2Z = Rand_CenteredFloat(10.0f) + player->actor.world.pos.z;
+                    }
+                    spawnedRockType = FIRE_ROCK_SPAWNED_FALLING2;
+                }
+            } else {
+                this->timerBetweenRockSpawns = 50;
+                spawnedRockType = FIRE_ROCK_SPAWNED_FALLING2;
+                // rock spawn pos X,Z at a random position roughly 300 units ahead of camera
+                tempVec2X = Rand_CenteredFloat(100.0f) + tempVec1X;
+                tempVec2Z = Rand_CenteredFloat(100.0f) + tempVec1Z;
+                if (Rand_ZeroFloat(3.99f) < 1.0f) {
+                    // rock spawn pos X,Z at a random position near player
+                    tempVec2X = Rand_CenteredFloat(70.0f) + player->actor.world.pos.x;
+                    tempVec2Z = Rand_CenteredFloat(70.0f) + player->actor.world.pos.z;
+                }
+            }
+            spawnedRock =
+                (EnFireRock*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_FIRE_ROCK,
+                                                tempVec2X, tempVec1Y, tempVec2Z, 0, 0, 0, spawnedRockType);
+            if (spawnedRock != NULL) {
+                spawnedRock->spawner = this;
+                this->numSpawnedRocks++;
+                return;
+            }
+            // "☆☆☆☆☆ Can't occur! ☆☆☆☆☆"
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n\n" VT_RST);
+        }
+    }
+}
+
+void EnEncount2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnEncount2* this = THIS;
+    GlobalContext* globalCtx2 = globalCtx;
```

Comment:
Usually we have this one be globalCtx and have globalCtx2 above, however this hasnt been standardized and I'd wait for more input on this.

Reply:
I've already changed it to your suggestion of globalCtx2 in args. If we decide later we don't like that then it can be changed back with all the other files that do that.